include/alpaka/mem/fence/Traits.hpp

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2021 Jan Stephan
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+#include <alpaka/core/Concepts.hpp>
+
+namespace alpaka
+{
+    struct ConceptMemFence
+    {
+    };
+
+    namespace memory_scope
+    {
+        //! Memory fences are observed by all threads in the same block.
+        struct Block
+        {
+        };
+
+        //! Memory fences are observed by all threads on the device.
+        struct Device
+        {
+        };
+    } // namespace memory_scope
```

Comment:
Do we also need a `System` scope? CUDA has `threadfence_system`.

Reply:
Nope, because we cannot support this for all back-ends. The problem: A CUDA system fence would also synchronize with memory used by OpenMP, but there is no way to issue an OpenMP fence and synchronize with GPU memory.

Reply:
ok!